ariadne/contrib/tracing/opentelemetry.py

```diff
@@ -0,0 +1,149 @@
+from functools import partial
+from inspect import iscoroutinefunction
+from typing import Any, Callable, Dict, Optional, Union
+
+from graphql import GraphQLResolveInfo
+from graphql.pyutils import is_awaitable
+from opentelemetry.trace import Context, Span, Tracer, get_tracer, set_span_in_context
+
+from ...types import ContextValue, Extension, Resolver
+from .utils import copy_args_for_tracing, format_path, should_trace
+
+
+ArgFilter = Callable[[Dict[str, Any], GraphQLResolveInfo], Dict[str, Any]]
+RootSpanName = Union[str, Callable[[ContextValue], str]]
+
+
+class OpenTelemetryExtension(Extension):
+    _arg_filter: Optional[ArgFilter]
+    _root_context: Optional[Context]
+    _root_span: Span
+    _root_span_name: Optional[RootSpanName]
+    _tracer: Tracer
+
+    def __init__(
+        self,
+        *,
+        tracer: Optional[Tracer] = None,
+        arg_filter: Optional[ArgFilter] = None,
+        root_context: Optional[Context] = None,
+        root_span_name: Optional[RootSpanName] = None,
+    ) -> None:
+        if tracer:
+            self._tracer = tracer
+        else:
+            self._tracer = get_tracer("ariadne")
+
+        self._arg_filter = arg_filter
+        self._root_context = root_context
+        self._root_span_name = root_span_name
+
+    def request_started(self, context: ContextValue):
+        if self._root_span_name:
+            if callable(self._root_span_name):
+                root_span_name = self._root_span_name(context)
+            else:
+                root_span_name = self._root_span_name
+        else:
+            root_span_name = "Anonymous GraphQL Operation"
+
+        span_context: Optional[Context] = None
+        if self._root_context:
+            span_context = self._root_context
+
+        root_span = self._tracer.start_span(root_span_name, context=span_context)
+        root_span.set_attribute("component", "GraphQL")
+        self._root_span = root_span
+
+    def request_finished(self, context: ContextValue):
+        self._root_span.end()
+
+    def resolve(
+        self, next_: Resolver, obj: Any, info: GraphQLResolveInfo, **kwargs
+    ) -> Any:
+        if not should_trace(info):
+            return next_(obj, info, **kwargs)
+
+        graphql_path = ".".join(
+            map(str, format_path(info.path))  # pylint: disable=bad-builtin
+        )
+
+        with self._tracer.start_as_current_span(
+            info.field_name, context=set_span_in_context(self._root_span)
+        ) as span:
+            span.set_attribute("component", "GraphQL")
+            span.set_attribute("graphql.parentType", info.parent_type.name)
+            span.set_attribute("graphql.path", graphql_path)
+
+            if info.operation.name:
+                span.set_attribute("graphql.operation", info.operation.name.value)
```

Comment:
```suggestion
                span.set_attribute("graphql.operation.name", info.operation.name.value)
```
per https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/instrumentation/graphql/